Sources/AlamofireXMLRPC/Alamofire+XML.swift

```diff
@@ -11,28 +11,63 @@ import AEXML
 import Alamofire
 
 extension DataRequest {
-    public static func XMLResponseSerializer() -> DataResponseSerializer<AEXMLDocument> {
-        return DataResponseSerializer { request, response, data, error in
-            if let e = error {
-                return .failure(e)
-            }
+    @discardableResult public func responseXML(
+        queue: DispatchQueue = .main,
+        dataPreprocessor: DataPreprocessor = XMLResponseSerializer.defaultDataPreprocessor,
+        emptyResponseCodes: Set<Int> = XMLResponseSerializer.defaultEmptyResponseCodes,
+        emptyRequestMethods: Set<HTTPMethod> = XMLResponseSerializer.defaultEmptyRequestMethods,
+        completionHandler: @escaping (AFDataResponse<AEXMLDocument>) -> Void
+    ) -> Self {
+        response(
+            queue: queue,
+            responseSerializer: XMLResponseSerializer(
+                dataPreprocessor: dataPreprocessor,
+                emptyResponseCodes: emptyResponseCodes,
+                emptyRequestMethods: emptyRequestMethods
+            ),
+            completionHandler: completionHandler)
+    }
+}
 
-            guard let validData = data else {
-                return .failure(AFError.responseValidationFailed(reason: AFError.ResponseValidationFailureReason.dataFileNil))
-            }
+public class XMLResponseSerializer: ResponseSerializer {
```

Comment:
Does it need to be a class? It seems to have no mutating state

Reply:
I just followed the convention in Alamofire. I'll see if I can refactor it to just be a function.

Reply:
A _struct_ should work, but I don't see any big issue keeping it as a class.

Reply:
Technically we could use a struct, the `DataResponseSerializerProtocol` only requires the `serialize` function, but this wouldn't be consistent with the other serializer implementations in Alamofire. Take a look at the `JSONResponseSerializer` (https://github.com/Alamofire/Alamofire/blob/master/Source/ResponseSerialization.swift#L651). I tend to favor consistency unless there's some constraint that says otherwise. What do you think?

Reply:
Ok, I find it strange but let's keep a class then.